doc/MSTP/MSTP.md

```diff
@@ -0,0 +1,1178 @@
+## CONFIG DB
+
+### Existing Table
+Following existing table of CONFIG_DB will be modified for MSTP implementation:
+
+#### STP_GLOBAL_TABLE
+A new value of `mst` for `mode` column
+```
+mode 		  = "pvst" / "mst"      ; a new option for mstp
+```
+Other fields of this table are not applicable for "mstp" mode.
+
+### New Tables
+Following new tables will be added to CONFIG_DB:
+
+
+#### STP_MST
+```
+;Stores STP Global configuration
+key                       = STP_MST| GLOBAL               ; GLOBAL MSTP table key 
+name                      = 1*32CHAR                      ; MSTP region name(DEF:mac address of switch)
+revision                  = 5*DIGIT                       ; MSTP revision number (0 o 65535 ,DEF:0)
+max_hop                   = 2*DIGIT                       ; maximum hops (1 to 40, DEF:20)   
+max_age                   = 2*DIGIT                       ; maximum age time in secs (6 to 40sec, DEF:20sec)
+hello_time                = 2*DIGIT                       ; hello time in secs (1 to 10sec, DEF:2sec)
+forward_delay             = 2*DIGIT                       ; forward_delay in secs (4 to 30sec, DEF:15 sec)
+```
+
+#### STP_MST_INST
+```
+;Stores STP configurations per MSTI
+key                 = STP_MST| "MST_INSTANCE":"Instance" id   ; MST with prefix "STP_MST"
+bridge_priority     = 5*DIGIT                                 ; bridge priority (0 to 61440, DEF: 32768)
+vlan_list           = "Vlans"                                 ; Lists of VLANs assigned to the MST instance
+```
+
+#### STP_MST_PORT 
+
+```
+;Stores STP interface details per MSTI
+key                 = STP_MST_PORT| "MST_INSTANCE" id|ifname    ; MSTI+Intf with prefix "STP_MST_PORT"
+path_cost           = 9*DIGIT                                   ; port path cost (1 to 20000000)
+priority            = 3*DIGIT                                   ; port priority (0 to 240, DEF:128)
+
+```
+
+#### STP_PORT
+
+```
+;Stores STP interface details
+key                 = STP_PORT|ifname              ; ifname with prefix STP_PORT
+edge_port           = BIT                          ; enabled or disabled
+link_type           = "type"                       ; type can be of auto, point-to-point or shared
+enabled             = BIT                          ; enabled or disabled
+bpdu_guard          = BIT                          ; enabled or disabled
+bpdu_guard_do       = BIT                          ; enabled or disabled 
+enabled             = BIT                          ; enabled or disabled
+root_guard          = BIT                          ; enabled or disabled 
+path_cost           = 9*DIGIT                      ; port path cost (1 to 20000000)
+priority            = 3*DIGIT                      ; port priority (0 to 240, DEF:128)
+
+```
+
+
+## APP DB
+
+### New Tables
+Following new tables are introduced as part of MSTP Feature:
+
+
+#### STP_MST_INST_TABLE
+```
+;Stores the STP per  MSTI operational details
+key                       = _STP_MST_INST_TABLE:"MST" id  
```

Comment:
As discussed in the meeting, remove "_" prefix from the tables.

Reply:
Removed